### DISCUSSION TITLE
Exporting models (both vanilla and modded)?

### DISCUSSION THREAD

**Discussion by nickgirga** [xivdev/Penumbra#151] [Q&A]
Sorry if it's inappropriate to ask this here, as I'm not really sure how else to acquire this information. If there's maybe a development Discord or some knowledgeable people in this subject I could poke the brains of, I'd really appreciate some help and/or pointers to some good documentation for this kind of thing.

(I've emboldened and italicized the most important parts of my inquiry because I know I talk too much)

**_I'm interested in exporting my entire character from FFXIV as a standard 3D model_** (e.g. fbx, obj, gltf). I know I could extract the meshes myself, stitch them together, and "shrinkwrap" the clothing, but that leaves a lot of room for error (mostly forgetting a few pieces or grabbing the wrong ones). So, I had the idea that I could maybe use a Dalamud plugin to automate the process of acquiring the necessary mesh files and applying the mesh modifiers for things like clothing that are likely sitting in memory as the game is running. But there are a few problems with this approach: I don't think there is an existing plugin that does this and I'm not sure how I would get it to work with modded items (using Penumbra).

I'm honestly mostly expecting that this type of plugin doesn't exist and I'm willing to write the plugin myself. But I've never created a Dalamud plugin before, so I'd like to know **_if any of this is even possible with the current set of tools Dalamud provides_** and **_if it's possible for my plugin to interact with Penumbra_** to achieve my desired results.

I've already asked in the XIVLauncher/Dalamud Discord, but they suggested this would be pretty difficult and told me that the TexTools and custom animations community might have some better information. I figured since almost everything I'm planning on doing pertains to Dalamud and Penumbra, it would be best to ask here.

Thanks in advance! Again, I apologize if this isn't the right place for this.

**Comment by slotthhy** [comment]
Textools FMV feature was built for this. Penumbra has no capability to do this.

**Reply by nickgirga** [reply]
I wasn't aware TexTools had that capability. But I'd still like to implement this as my own plugin, separate from Penumbra if possible. TexTools doesn't work on my system and I prefer Penumbra anyways, so I'd rather not switch.

Is this even possible with the current set of tools that Dalamud provides?

**Reply by slotthhy** [reply]
I say this as someone who has attempted this exact thing before, just use Textools in any way you can. A Dalamud plugin is not suitable for this use case. You need in-depth knowledge of how the game resolves models, materials, how it hides parts, when it hides parts, which parts it hides, how models define their parts, how the skeleton gets deformed, how to extract the skeleton, it's just not worth the time. Use Textools.

**Reply by nickgirga** [reply]
That's a shame. I'd like to create a permanent solution for people using Penumbra.

Anyways, thanks for your time. I appreciate it.
